source/features/ci-link.tsx

```diff
@@ -54,22 +56,74 @@ async function init(signal: AbortSignal): Promise<void> {
 		// Old selector: `.avatar` excludes "Global navigation update"
 		// Repo title (aware of forks and private repos)
 		'[itemprop="name"]:not(.avatar ~ [itemprop])',
-	], add, {signal});
+	], addRepoIcon, {signal});
+}
+
+function addPRIcon(base?: DocumentFragment | HTMLElement): void {
+	const selectBase = base instanceof DocumentFragment ? base : document.documentElement;
+	const iconWrapper = lastElement('.TimelineItem .js-socket-channel[data-url]', selectBase);
+	if (!iconWrapper) {
+		return;
+	}
+
+	$('#partial-discussion-header')!.classList.add('rgh-pr-ci-link-added');
+
+	const ciLinkCommitSha = /[a-f\d]{40}/.exec(iconWrapper.dataset.url!)![0];
+	const prTitles = $$(`.js-issue-title:not([data-rgh-ci-link-commit="${ciLinkCommitSha}"])`); // Avoid duplicating CI link
+	for (const title of prTitles) {
+		if (title.dataset.rghCiLinkCommit) {
+			title.lastElementChild!.remove(); // Drop outdated CI link
+		}
+
+		title.dataset.rghCiLinkCommit = ciLinkCommitSha;
+		title.parentElement!.append(iconWrapper.cloneNode(true));
+	}
+}
+
+async function fetchAndAddPrIcon(): Promise<void> {
+	const prCommitListURL = buildRepoURL('pull', getConversationNumber()!) + '?' + String(Math.random()); // Break memoization to always fetch the latest DOM
+	addPRIcon(await fetchDom(prCommitListURL));
+}
+
+async function initPR(signal: AbortSignal): Promise<void> {
+	const ciLinkIsInDom = pageDetect.isPRConversation() || pageDetect.isPRCommitList();
+
+	if (ciLinkIsInDom) {
+		observe(':is(#discussion_bucket, #commits_bucket) .commit-build-statuses summary', addPRIcon, {signal});
```

Comment:
Note that this will trigger for each and every `check`ed-commit on the page. You might just want to use `awaitDomReady` and `lastElement` instead, even if it's slightly longer.

However I'd probably just skip this and always use `fetchAndAddPrIcon` for simplicity.